pennylane_lightning/src/tests/TestHelpers.hpp

```diff
@@ -9,49 +9,78 @@
 #include "TestKernels.hpp"
 #include "Util.hpp"
 
-#include <catch2/catch.hpp>
+#include <catch2/catch_all.hpp>
 
 #include <algorithm>
 #include <complex>
 #include <random>
 #include <string>
 #include <type_traits>
+#include <variant>
 #include <vector>
 
 namespace Pennylane {
 template <class T, class Alloc = std::allocator<T>> struct PLApprox {
-    const std::vector<T, Alloc> &comp_;
+    using param_t = std::variant<std::vector<T, Alloc>, T, std::monostate>;
```

Comment:
Hi @mlxd, I indeed wanted to change the structure of this class using Catch2 v3's new matcher https://github.com/catchorg/Catch2/blob/devel/docs/matchers.md#writing-custom-matchers-new-style. In fact, I have implemented one for MrMustard in https://github.com/XanaduAI/MrMustard/blob/add_cpp_math/cpp/tests/test_helper.hpp#L35-L80. Can I jump in this PR and change this class? Or you can just remain the minimal working class here and I will make another PR.